File: app/src/main/java/com/example/proyeksp/helper/CurrencyHelper.java
package com.example.proyeksp.helper;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.text.NumberFormat;
import java.util.Currency;
import java.util.Locale;

public class CurrencyHelper {
    private static NumberFormat format;

    public static String format(long n) {
        if (format == null) {
            format = NumberFormat.getCurrencyInstance(new Locale("id", "ID"));
            format.setCurrency(Currency.getInstance("IDR"));
            format.setMaximumFractionDigits(0);
            format.setGroupingUsed(true);
        }

        return format.format(n);
    }
}

File: app/src/main/java/com/example/proyeksp/helper/CurrencyHelper.kt
package com.example.proyeksp.helper

import java.text.NumberFormat
import java.util.Currency
import java.util.Locale

object CurrencyHelper {
    private val format: NumberFormat by lazy {
        NumberFormat.getCurrencyInstance(Locale("in", "ID")).apply {
            currency = Currency.getInstance("IDR")
            maximumFractionDigits = 0
            isGroupingUsed = true
        }
    }

    fun format(n: Long): String {
        return format.format(n)
    }
}